pkg/hive/cell/health.go

```diff
@@ -57,6 +58,25 @@ type HealthReporter interface {
 	Degraded(reason string, err error)
 }
 
+// Update represents an instantaneous health status update.
+type Update interface {
+	// Level returns the level of the update.
+	Level() Level
+
+	// String returns a string representation of the update.
+	String() string
+
+	// JSON returns a JSON representation of the update, this is used by the agent
+	// health CLI to unmarshal health status into cell.StatusNode.
+	JSON() ([]byte, error)
```

Comment:
Should we mix in json.Marshaller instead?

Reply:
I tried that, the UnmarshalJSON function on recursive data structure tends to cause problems 😉